pkg/kafka/kafka-publisher.go

```diff
@@ -202,8 +200,8 @@ func NewKafkaPublisher(kafkaSrv string) (pub.Publisher, error) {
 	}, nil
 }
 
-func validator(addr string) error {
-	host, port, _ := net.SplitHostPort(addr)
+func validator(kConfig Config) error {
```

Comment:
Please pass kConfig by reference and check for nil to prevent panic when dereferencing..